test_that_surfaces_retain_their_order_when_loaded_and_saved_by_ert is flaky

**Describe the bug**
Reported by @eivindjahren 

Flaky on 3.11
```
FAILED tests/integration_tests/analysis/test_es_update.py::test_that_surfaces_retain_their_order_when_loaded_and_saved_by_ert[False] - AssertionError: 
Arrays are not almost equal to 3 decimals

Mismatched elements: 6 / 35 (17.1%)
Max absolute difference: 0.00208282
Max relative difference: 4.5674897e-05
 x: array([[14.62 , 16.089, 18.916, 22.657, 26.425, 29.272, 30.716],
       [14.915, 16.587, 19.801, 24.016, 28.187, 31.259, 32.78 ],
       [15.486, 17.523, 21.448, 26.573, 31.551, 35.104, 36.805],...
 y: array([[14.621, 16.089, 18.917, 22.658, 26.426, 29.273, 30.718],
       [14.915, 16.588, 19.801, 24.017, 28.188, 31.26 , 32.781],
       [15.486, 17.524, 21.449, 26.574, 31.552, 35.106, 36.806],...
```

--- a/tests/integration_tests/analysis/test_es_update.py
+++ b/tests/integration_tests/analysis/test_es_update.py
@@ -186,7 +186,7 @@ def test_that_surfaces_retain_their_order_when_loaded_and_saved_by_ert(
         with pytest.raises(AssertionError):
             np.testing.assert_array_equal(surf_prior[i], surf_posterior[i])
         np.testing.assert_almost_equal(
-            surf_prior[i].values, surf_posterior[i].values, decimal=3
+            surf_prior[i].values, surf_posterior[i].values, decimal=2
         )
 
 

